FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Keep_Alive/DemoTasks/KeepAliveMQTTExample.c

```diff
@@ -917,19 +969,39 @@ static void prvMQTTProcessIncomingPublish( MQTTPublishInfo_t * pxPublishInfo )
 
 /*-----------------------------------------------------------*/
 
-static void prvKeepAliveTimerCallback( TimerHandle_t pxTimer )
+static void prvPingReqTimerCallback( TimerHandle_t pxTimer )
 {
     TransportInterface_t * pxTransport;
     int32_t xTransportStatus;
+    BaseType_t xTimerStatus;
 
     pxTransport = ( TransportInterface_t * ) pvTimerGetTimerID( pxTimer );
 
-    /* Send Ping Request to the broker. */
-    LogInfo( ( "Ping the MQTT broker." ) );
-    xTransportStatus = pxTransport->send( pxTransport->pNetworkContext,
-                                          ( void * ) xPingReqBuffer.pBuffer,
-                                          xPingReqBuffer.size );
-    configASSERT( ( size_t ) xTransportStatus == xPingReqBuffer.size );
+    /* Do not resend if waiting on a PINGRESP. */
+    if( xWaitingForPingResp == false )
+    {
+        /* Send PINGREQ to broker */
+        LogInfo( ( "Ping the MQTT broker." ) );
+        xTransportStatus = pxTransport->send( pxTransport->pNetworkContext,
+                                              ( void * ) xPingReqBuffer.pBuffer,
+                                              xPingReqBuffer.size );
+        configASSERT( ( size_t ) xTransportStatus == xPingReqBuffer.size );
+
+        xWaitingForPingResp = true;
+        /* Start the timer to expect a PINGRESP. */
+        xTimerStatus = xTimerStart( xPingRespTimer, 0 );
```

Comment:
Is it fine to be calling start on this timer repeatedly without calling stop on it?

Reply:
The third parameter of `xTimerCreateStatic` allows you to specify whether the time triggers periodically or just once. For the PINGRESP, once it's trigged, it won't start again unless you do so again.